hw_4_js/exercises.js

```diff
@@ -0,0 +1,415 @@
+ * [1, 2, 2, 4, 5, 5] => [1, 2, 4, 5]
+ *
+ * Use: reduce and indexOf
+ */
+
+const getUniqueValues = (numbers) => {
+  return numbers.reduce(
+    (uniques, current) => uniques.indexOf(current) < 0 ? uniques.concat(current) : uniques,
+    []
+  )
+}
+
+/**
+ * Exercise 11
+ *
+ * Implement a function whose input is a numeric code of an error, the output is a string with a message
+ * 500 => Server Error
+ * 401 => Authorization failed
+ * 402 => Server Error
+ * 403 => Access denied
+ * 404 => Not found
+ *
+ * Use: switch case or object like a map structure
+ */
+
+const getErrorMessage = (code) => {
+  switch (code) {
+    case 500:
+      return 'Server Error'
+    case 401:
+      return 'Authorization failed'
+    case 402:
+      return 'Server Error'
+    case 403:
+      return 'Access denied'
+    case 404:
+      return 'Not found'
+    default:
+      return 'Wrong status code'
+  }
+}
+
+/**
+ * Exercise 12
+ *
+ * Write a function that returns the 2 smallest values of an array
+ * [4,3,2,1] => [1,2]
+ *
+ * Use: .sort()
+ */
+
+const get2SmallestValues = (numbers) => {
+  return numbers.sort().slice(0, 2)
+}
+
+/**
+ * Exercise 13
+ *
+ * Implement a function, at the input of which an object of the following form:
+ * {
+ * firstName: 'Peter',
+ * secondName: 'Vasiliev',
+ * patronymic: 'Ivanovich'
+ *}
+ * output line with the message 'Name: Petr Ivanovich Vasiliev'
+ */
+
+const getFullName = (user) => {
+  return 'Name: ' + user.firstName + ' ' + user.patronymic + ' ' + user.secondName
+}
+
+/**
+ * Exercise 14
+ *
+ * Implement a function that takes 2 arguments as input: an array of numbers and a multiplier,
+ * a returns an array of the original array, each element of which has been multiplied by a factor:
+ *
+ * [1,2,3,4], 5 => [5,10,15,20]
+ *
+ * Use: map
+ */
+
+const multiplyTo = (numbers, multiplier) => {
+  return numbers.map(n => n * multiplier)
+}
+
+/**
+ * Exercise 15
+ *
+ * Implement a function that takes 2 arguments as input: an array and a franchise,
+ * and returns a string with the names of the heroes separated by a comma:
+ *
+ *[
+ * {name: "Batman", franchise: "DC"},
+ * {name: "Ironman", franchise: "Marvel"},
+ * {name: "Thor", franchise: "Marvel"},
+ * {name: â€œSupermanâ€, franchise: â€œDCâ€}
+ *],
+ * Marvel
+ * => Ironman, Thor
+ *
+ * Use: filter, map, join
+ */
+
+const getCharacterNames = (characters, franchise) => {
+  return characters
+    .filter(c => c.franchise == franchise)
+    .map(c => c.name)
+    .join(', ')
+}
+
+// ----==== Advanced exercises (8 items) ====----
+/**
+ * Exercise 16
+ *
+ * Write a function that returns an array of the smallest row values of a two-dimensional array
+ *[
+ * [10,1,300,4],
+ * [20,2,300,400],
+ * [30,3,300,4],
+ * [40,4,300,4],
+ *]
+ * => [1,2,3,4]
+ */
+
+const getSmallestRow = (numbers) => {
+  return numbers.reduce((acc, current) => acc.concat([Math.min(...current)]), [])
```

Comment:
`numbers.map(...)` is enough

Reply:
I agree, it seems to be overcomplicated, fixed